iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ThreeWindingsTransformerImpl.java

```diff
@@ -386,24 +386,30 @@ public Terminal getTerminal(ThreeSides side) {
     @Override
     public Terminal getTerminal(String voltageLevelId) {
         Objects.requireNonNull(voltageLevelId);
-        boolean side1 = getLeg1().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
-        boolean side2 = getLeg2().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
-        boolean side3 = getLeg3().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
-        if (side1 && side2 && side3) {
+        boolean isLeg1ConnectedToVoltageLevel = isLegOnVoltageLevel(getLeg1(), voltageLevelId);
+        boolean isLeg2ConnectedToVoltageLevel = isLegOnVoltageLevel(getLeg2(), voltageLevelId);
+        boolean isLeg3ConnectedToVoltageLevel = isLegOnVoltageLevel(getLeg3(), voltageLevelId);
+        if (isLeg1ConnectedToVoltageLevel && isLeg2ConnectedToVoltageLevel && isLeg3ConnectedToVoltageLevel) {
             throw new PowsyblException("The three terminals are connected to the same voltage level " + voltageLevelId);
-        } else if (side1 && side2 || side3 && side1 || side2 && side3) {
+        } else if (isLeg1ConnectedToVoltageLevel && isLeg2ConnectedToVoltageLevel || isLeg3ConnectedToVoltageLevel && isLeg1ConnectedToVoltageLevel || isLeg2ConnectedToVoltageLevel && isLeg3ConnectedToVoltageLevel) {
```

Comment:
With the variables' renaming, I find the code less readable.
Can you rename the variables with shorter names (for instance `vlOnSide1`, ...), or at least split this condition before each `|`?